### DISCUSSION TITLE
A canvas like element?

### DISCUSSION THREAD

**Discussion by jl-a** [slint-ui/slint#1080] [Q&A] [answered]
I have image data that I'm manipulating per pixel, and am looking to visualise it as it's edited. Is it possible to display the image data in Slint without using the filesystem? A canvas element like HTML5's canvas would be ideal.

**Comment by tronical** [comment, marked as answer]
If you're using Rust, you can use the [`slint::Image`](https://slint-ui.com/releases/0.2.1/docs/rust/slint/struct.image) type to construct a copy of your pixel buffer to draw it in a `Image` element.

Or are you using C++ by chance?

**Comment by jl-a** [comment]
Yes, I'm using Rust. Thank you, that was very helpful! I've found the docs for `slint::Image`.
